src/clients/SpokePoolClient/SpokePoolClient.ts

```diff
@@ -960,17 +753,75 @@ export class SpokePoolClient extends BaseAbstractClient {
     );
   }
 
+  // ///////////////////////
+  // // ABSTRACT METHODS //
+  // ///////////////////////
+
+  /**
+   * Returns a list of event names that are queryable for the SpokePoolClient.
+   * @returns A list of event names that are queryable for the SpokePoolClient.
+   */
+  public abstract _queryableEventNames(): string[];
+
+  /**
+   * Performs an update to refresh the state of this client. This will query the SpokePool contract for new events
+   * and store them in memory. This method is the primary method for updating the state of this client.
+   * @param eventsToQuery An optional list of events to query. If not provided, all events will be queried.
+   * @returns A Promise that resolves to a SpokePoolUpdate object.
+   */
+  protected abstract _update(eventsToQuery: string[]): Promise<SpokePoolUpdate>;
+
+  /**
+   * Finds the deposit id at a specific block number.
+   * @param blockTag The block number to search for the deposit ID at.
+   * @returns The deposit ID.
+   */
+  public abstract _getDepositIdAtBlock(blockTag: number): Promise<BigNumber>;
+
```

Comment:
It should be fine to drop this, per the above comment :+1: 

```suggestion
```

Reply:
https://github.com/across-protocol/sdk/pull/935